Add WIF documentation for Azure Auth and Secrets engines (#27185)
hashicorp/vault: website/content/api-docs/auth/azure.mdx

to Azure. These credentials will be used to query the metadata about the
virtual machine.

> Credentials can be configured either with account credentials, or using
> Plugin Workload Identity Federation (WIF).

Reviewer: ```suggestion
You can configure the API credentials with account credentials or plugin workload
 identity federation (WIF).
```
Style correction: write in active voice
